src/cluster_undersample/cluster_undersample.jl

```diff
@@ -0,0 +1,207 @@
+"""
+Assuming that all the observations in the observation matrix X belong to the same class, undersample
+the observations using clustering undersampling
+
+# Arguments
+- `X`: A matrix where each column is an observation of floats
+- `n`: Number of observations in the undersampled dataset
+- `mode::AbstractString="nearest`: If `center` then the undersampled data will consist of the centriods of 
+    each cluster found. Meanwhile, if `nearest` then it will consist of the nearest neighbor of each centroid.
+- `maxiter::Integer=100`: Maximum number of iterations to run K-means
+- `rng::Integer=42`: Random number generator seed
+# Returns
+- `Xnew`: A matrix containing the undersampled observations
+"""
+function cluster_undersample_per_class(
+    X::AbstractMatrix{<:Real},
+    n::Integer;
+    mode::AbstractString = "nearest",
+    maxiter::Integer = 100,
+    rng::Integer = 42,
+)
+    (mode in ["center", "nearest"]) ||
+        throw(ArgumentError("mode must be either 'center' or 'nearest'"))
```

Comment:
Better maintenance and testing to define the exception as a constant, as I think we already to elsewhere, no?

Reply:
Yes, correct. Fixed it and another instance.